spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/resolver/AetherModuleResolver.java

```diff
@@ -169,4 +154,100 @@ public void serviceCreationFailed(Class<?> type, Class<?> impl, Throwable except
 		});
 		return locator.getService(RepositorySystem.class);
 	}
+
+
+	/**
+	 * Resolve a set of artifacts based on their coordinates, including their dependencies, and return the locations of
+	 * the transitive set in the local repository. Aether performs the normal Maven resolution process ensuring that the
+	 * latest update is cached to the local repository. A number of additional includes and excludes can be specified,
+	 * allowing to override the transitive dependencies of the original set. Includes and their transitive dependencies
+	 * will always
+	 *
+	 * @param coordinates the Maven coordinates of the artifacts
+	 * @return a {@link FileSystemResource} representing the resolved artifact in the local repository
+	 * @throws RuntimeException if the artifact does not exist or the resolution fails
+	 */
+	@Override
+	public Resource[] resolve(Coordinates[] coordinates, Coordinates[] includes, String[] excludePatterns) {
+		Assert.notEmpty(coordinates, "A list of root nodes must be provided");
+		for (Coordinates coordinate : coordinates) {
+			validateCoordinates(coordinate);
+		}
+		if (!ObjectUtils.isEmpty(includes)) {
+			for (Coordinates include : includes) {
+				validateCoordinates(include);
+			}
+		}
+		List<Resource> result = new ArrayList<>();
+		// if we only have one artifact to resolve, it will be set as root
```

Comment:
To note, I will change the aggregation handling to use the first module as root and the next ones as  direct dependencies. While not entirely identical to having all as direct dependencies, in practice the end result of the resolution process should be the same (and if there are inherent conflicts then they should be resolved as part of the exclusion process).